nrf-mpsl/src/mpsl.rs

```diff
@@ -94,9 +100,23 @@ impl<'d> MultiprotocolServiceLayer<'d> {
         TIMER0::set_priority(Priority::P0);
         POWER_CLOCK::set_priority(Priority::P4);
 
-        Ok(MultiprotocolServiceLayer { _private: PhantomData })
+        Ok(Self { _private: PhantomData })
+    }
+
+    pub fn timeslot<const N: usize>(self, slots: u8, mem: &'d mut Mem<N>) -> Result<Self, Error> {
+        let ret = unsafe { raw::mpsl_timeslot_session_count_set(mem.0.as_mut_ptr() as *mut _, slots) };
+        RetVal::from(ret).to_result()?;
+        Ok(self)
+    }
```

Comment:
The timeslot session memory has different requirements from the SDC memory, so I think this should be a different type. For example:
```rust
#[repr(align(4))]
pub struct SessionMem<const N: usize>(MaybeUninit<[[u8; raw::MPSL_TIMESLOT_CONTEXT_SIZE]; N]>);
```

You can then get `slots` directly from the type.

I also wonder if it would be better to just make a `MultiprotocolServiceLayer::with_timeslots` constructor rather than a builder. Are there any other initialization-type configurations we will need to add later?

Reply:
Separate memory sounds good to me. About the constructor, it felt consistent with the SoftdeviceController, and I think maybe with a proper timeslot API there might be other options to consider (but I haven't really checked).

Reply:
I haven't seen any other init-time settings, but I haven't looked in depth either.

Reply:
Fixed